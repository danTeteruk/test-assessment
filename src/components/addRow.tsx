import { inject, observer } from "mobx-react";
import React, { FC, useCallback, useState } from "react";
import { ToDo, ToDoStore } from "../stores/ToDoStore";

type InjectedProps = {
  toDoStore: ToDoStore;
};

export const AddRow: FC<InjectedProps> = ({ toDoStore }) => {
  const [value, setValue] = useState("");

  const onAddButtonClick = useCallback((): void => {
    if (!value) return;
    toDoStore.addToDo(value);
    setValue("");
  }, [toDoStore, value, setValue]);

  return (
    <div className="add">
      <input onChange={(e) => setValue(e.target.value)} value={value} />
      <button onClick={onAddButtonClick}>Add</button>
    </div>
  );
};

type ListProps = {
  todos: ToDo[];
  toggleItem: (id: number) => void;
};

export const List: FC<ListProps> = observer(({ todos, toggleItem }) => {
  return (
    <>
      {todos.map((item: ToDo) => (
        <div className={`item ${item.isDone && "crossed"}`} key={item.id}>
          <div>{item.title}</div>
          <div data-testid="toggle"
            className={`checkbox ${item.isDone && "checked"}`}
            onClick={() => toggleItem(item.id)}
          ></div>
        </div>
      ))}
    </>
  );
});

export default inject("toDoStore")(observer(AddRow)) as unknown as FC;
