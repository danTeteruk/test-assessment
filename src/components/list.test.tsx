import { cleanup, fireEvent, render } from "@testing-library/react";
import React from "react";
import { List } from "./addRow";

afterEach(cleanup);

describe("UrlRedirectForm", () => {
  it("renders and matches snapshot", async () => {
    const { asFragment, container } = render(
      <List
        todos={[{ id: 1, title: "some titile", isDone: false }]}
        toggleItem={jest.fn()}
      />
    );

    expect(container).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it("triggers toggleItem callback", async () => {
    const mockToggle = jest.fn();

    const { getByTestId } = render(
      <List
        todos={[{ id: 1, title: "some titile", isDone: false }]}
        toggleItem={mockToggle}
      />
    );

    fireEvent.click(getByTestId("toggle"))
    expect(mockToggle).toHaveBeenCalledWith(1);
  });
});
