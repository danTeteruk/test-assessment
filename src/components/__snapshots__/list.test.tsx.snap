// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlRedirectForm renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="item false"
  >
    <div>
      some titile
    </div>
    <div
      class="checkbox false"
      data-testid="toggle"
    />
  </div>
</DocumentFragment>
`;
